github_nonpublic_api/api.py

```diff
@@ -25,6 +29,7 @@ def _get_and_submit_form(session, url: str, data_callback=None, form_matcher=lam
             submit_form = form
             break
     if submit_form is None:
+        logging.error('Unable to find form at URL %s', url)
```

Comment:
So in general, we don't want to both log something and throw an exception (this goes for python and java as well).  Typically we want to catch this exception at a higher level and add the logging there.

The reason we don't want to do the logging here is because it will generally get logged at the higher level as well, so we would get double prints.

If we want to bubble the url up as part of the exception message, that would be a nice change.